client/my-sites/controller.js

```diff
@@ -145,26 +145,26 @@ function isPathAllowedForDomainOnlySite( pathname, domainName ) {
 
 function onSelectedSiteAvailable( context ) {
 	const selectedSite = sites.getSelectedSite();
-	const state = context.store.getState();
+	const getState = () => context.store.getState();
```

Comment:
This shouldn't be needed, you get a reference to the store here so even if `setSelectedSiteId` modifies it you'll be able to read the updated value